plugins/statistics.koplugin/main.lua

```diff
@@ -929,7 +937,7 @@ function ReaderStatistics:getReadingProgressStats(sdays)
         ORDER  BY dates DESC
     ]]
     local result_book = conn:exec(string.format(sql_stmt, period_begin))
-
+    if result_book == nil then return nil end
```

Comment:
If you just return then you always return nil. :-) Btw, is it important to check for nil or could it be simplified to `if not result_book then return end`?